Added interactor that observes the VLVisibilityRepository (added) and displays the VLs through its displayer.

teamagam5 opened the pull request:
Added interactor that observes the VLVisibilityRepository (added interface and impl.) and displays the VLs through its displayer (interface).
Tests are included.

Impl. note: Introduced VLVisibilityChange (similar to GeoEntityNotification) in order to track current status.

teamagam5 committed:
Added interactor that observes the VLVisibilityRepository (added interface and impl.) and displays the VLs through its displayer (interface).
Tests are included.

Impl. note: Introduced VLVisibilityChange (similar to GeoEntityNotification) in order to track current status.

teamagam5 committed:
Separation between VL *visibility* repo and the VL repo itself.
Now the visibility repo has only IDs.

teamagam3 commented on data/src/main/java/com/teamagam/gimelgimel/data/map/repository/VectorLayersDataRepository.java:
```diff
@@ -0,0 +1,56 @@
+        if (!mIdToVectorLayersMap.containsKey(vectorLayer.getId())) {
+            mIdToVectorLayersMap.put(vectorLayer.getId(), vectorLayer);
+            mVectorLayerInnerRepo.add(vectorLayer);
+        }
```
Add an error indication or do an update when the you already have the layer.

teamagam3 commented on data/src/main/java/com/teamagam/gimelgimel/data/map/repository/VectorLayersDataRepository.java:
```diff
@@ -0,0 +1,56 @@
+
+    @Override
+    public VectorLayer update(VectorLayer vectorLayer) {
+        VectorLayer removed = remove(vectorLayer.getId());
```
If you get an invalid object or object with id you don't have, you'll get null object.
Fix it.

teamagam3 commented on domain/src/main/java/com/teamagam/gimelgimel/domain/map/DisplayVectorLayersInteractor.java:
```diff
@@ -0,0 +1,55 @@
+            DisplaySubscriptionRequest.DisplaySubscriptionRequestFactory factory) {
+        return factory.create(
+                mVectorLayersVisibilityRepository.getVisibilityChangesLogObservable(),
+                visibilityChange -> display(
```
Extract to function or something that will be easier to read.
I've found it hard to understand.

teamagam5 committed:
CR Fixes.
Removed functionality from the VL Repo interface (the methods remove and update).
